services/supervisord/supervisord.go

```diff
@@ -406,6 +422,45 @@ func (s *Service) marshalConfig(tmpl *template.Template, settings *models.Settin
 	return b, nil
 }
 
+// addAlertManagerParam parses alertManagerURL
+// and extracts username and password from it to templateParams.
+func addAlertManagerParam(alertManagerURL string, templateParams map[string]interface{}) error {
+	templateParams["AlertmanagerURL"] = alertManagerURL
+	templateParams["AlertManagerUser"] = ""
+	templateParams["AlertManagerPassword"] = ""
+	if alertManagerURL == "" {
+		return nil
+	}
+	u, err := url.Parse(alertManagerURL)
+	if err != nil {
+		templateParams["AlertmanagerURL"] = ""
+
+		return errors.Wrap(err, "cannot parse AlertManagerURL")
+	}
+
+	if u.Opaque != "" || u.Host == "" {
+		templateParams["AlertmanagerURL"] = ""
+
+		return errors.Errorf("AlertmanagerURL parsed incorrectly as %#v", u)
+	}
+	if username := u.User.Username(); username != "" {
+		if password, ok := u.User.Password(); ok {
+			n := url.URL{
```

Comment:
Any reason for this? What will change?

Reply:
extract username and password, it will prevent `vmaler`t to print it out to log file and flag value as plaintext.

 This behavior was fixed at last release.

Reply:
So we can just set empty values to Username and Password of `u`, since we don't use `u` anymore, right? to make it clear for everybody

Reply:
Yep, exactly. It's also possible to use username and password within url. But it's less secure and dont recommend.